### PR TITLE
:bug: Limit reset overrides on swapped components to affected subtree

### PR DESCRIPTION
**Pull request by superalex**
https://tree.taiga.io/project/penpot/issue/7254

### CODE REVIEW

File: frontend/src/app/main/data/workspace/libraries.cljs [hunk 1]
```diff
@@ -706,6 +706,38 @@
 
          (rx/take-until stopper-s))))))
 
+(defn sync-head
+  [id]
+  (ptk/reify ::sync-head
+    ptk/WatchEvent
+    (watch [it state _]
+      (log/info :msg "RESET-head of shape" :id (str id))
```
Comment (hirunatan) [inline frontend/src/app/main/data/workspace/libraries.cljs]: SYNC-head ?

File: frontend/src/app/main/data/workspace/libraries.cljs [hunk 2]
```diff
@@ -706,6 +706,38 @@
...
+            components-v2
+            (features/active-feature? state "components/v2")
+
+            changes
+            (-> (pcb/empty-changes it)
+                (pcb/with-container container)
+                (pcb/with-objects (:objects container))
+                (dwlh/generate-sync-shape-direct file-full libraries container (:id head) false components-v2))]
+
+        (log/debug :msg "RESET-head finished" :js/rchanges (log-changes
```
Comment (hirunatan) [inline frontend/src/app/main/data/workspace/libraries.cljs]: here too

File: common/src/app/common/types/component.cljc [hunk 3]
```diff
@@ -138,10 +138,11 @@
...
 
 (defn is-main-of?
-  [shape-main shape-inst]
+  [shape-main shape-inst components-v2]
   (and (:shape-ref shape-inst)
-       (or (= (:shape-ref shape-inst) (:id shape-main))
-           (= (:shape-ref shape-inst) (:shape-ref shape-main)))))
+       (cond-> (= (:shape-ref shape-inst) (:id shape-main))
+         (not components-v2)
+         (= (:shape-ref shape-inst) (:shape-ref shape-main)))))
```
Comment (hirunatan) [inline common/src/app/common/types/component.cljc]: I think this way would be clearer:

```clojure
       (or (= (:shape-ref shape-inst) (:id shape-main))
           (and (= (:shape-ref shape-inst) (:shape-ref shape-main))
                (not components-v2)))))
```

Comment (superalex) [reply common/src/app/common/types/component.cljc]: Thanks!

**Pull request merged**